packages/alpinejs/src/magics/$watch.js

```diff
@@ -1,26 +1,87 @@
 import { magic } from '../magics'
 
 magic('watch', (el, { evaluateLater, effect }) => (key, callback) => {
+    if (!window.structuredClone) {
+        console.warn('structuredClone not available on Window object. A polyfill is needed for full watch functionality.')
+    }
+
     let evaluate = evaluateLater(key)
 
     let firstTime = true
 
     let oldValue
 
-    let effectReference = effect(() => evaluate(value => {
-        // JSON.stringify touches every single property at any level enabling deep watching
-        JSON.stringify(value)
+    const shallowClone = (obj) => {
+        if (typeof obj !== 'object') {
```

Comment:
Hi all — just implemented this PR in my own project and got a JS error for my troubles. It was due to this line assuming that if `obj` is an object, then it can be parsed by Object.keys — This is incorrect, as `null` is also an object, but cannot be parsed:

```
Object.keys(null)
> Uncaught TypeError: can't convert null to object
``` 

I would recommend you add another check for `null`, e.g:

```
if (
    typeof obj !== 'object' ||
    obj === null
) {
```

(There may be other clauses needed here but this one got it working for me).

Also: Is it odd that it tested? Apologies, but I didn't have time to look at your test suites.

Reply:
(Not doing this as an official review as I don't know your contrib process. Sorry if I'm stepping on toes here!)

Reply:
Apologies guys but I've also noticed that the watcher only seems to detect the change in value. No subsequent changes are detected. I've not got a lot of time to look into this properly but thought you should know regardless. Thanks!